### PR TITLE
Cleanup: share hash tags newtypes in codebase/parser-typechecker

### PR DESCRIPTION
**Pull request by mitchellwrosen**
## Overview

Previously, we had a `U.Codebase.HashTags` module in `unison-codebase`, and some equivalent newtypes defined in `unison-parser-typechecker`.

This PR unifies them by moving `U.Codebase.HashTags` to a shared dependency, `unison-core`.

Also, while I was there, I noticed two unused hash newtype wrappers (`EditHash` and `DefnHash`), so I deleted them.

### CODE REVIEW

File: parser-typechecker/src/Unison/Codebase/SqliteCodebase/Operations.hs [hunk 1]
```diff
@@ -554,32 +553,22 @@ referentsByPrefix doGetDeclType (SH.ShortHash prefix (fmap Cv.shortHashSuffix1to
...
   -- refer to to specify a full namespace w/ history.
   -- but do we want to be able to refer to a namespace without its history?
-  cs <- Ops.causalHashesByPrefix (Cv.sch1to2 sh)
-  pure $ Set.map (Causal.CausalHash . unCausalHash) cs
-
--- returns `Nothing` to not implemented, fallback to in-memory
---    also `Nothing` if no LCA
--- The result is undefined if the two hashes are not in the codebase.
--- Use `Codebase.lca` which wraps this in a nice API.
-sqlLca :: Branch.CausalHash -> Branch.CausalHash -> Transaction (Maybe Branch.CausalHash)
```
Comment (ChrisPenner) [inline parser-typechecker/src/Unison/Codebase/SqliteCodebase/Operations.hs:569]: Did you mean to delete this? Presumably it's unused, but seems unrelated to just removing the extra set of hashes.

**Review by ChrisPenner — APPROVED** [review]
Noticed one function got deleted that I was curious about, but otherwise LGTM.

The two kinds of hashes problem has been a mild annoyance to me for a while, thanks!

File: parser-typechecker/src/Unison/Codebase/SqliteCodebase/Operations.hs (hunk 1 above)
Comment (mitchellwrosen) [reply parser-typechecker/src/Unison/Codebase/SqliteCodebase/Operations.hs:569]: Yeah, I did mean to delete it - after unifying the two `CausalHash` types it just became `sqlLca = lca` (from the other `Operations` module)
